lib/bridge/MatrixHandler.js

```diff
@@ -369,7 +369,101 @@ MatrixHandler.prototype._onAdminMessage = Promise.coroutine(function*(req, event
         return;
     }
     else {
-        req.log.info("No valid admin command: %s", event.content.body);
+        let fmt = '\nirc.server\n'+
+                    '(COMMAND [arg0 [arg1 [...]]])\n'+
+                    '(COMMAND [arg0 [arg1 [...]]])\n'+
+                    'irc.server2';
+        req.log.info(`No valid (old form) admin command, will try new format: ${fmt}`);
+
+        // Assumes commands have the form
+        // irc.server
+        // (COMMAND [arg0 [arg1 [...]]])
+        // (COMMAND [arg0 [arg1 [...]]])
+        // irc.server2
+        // ...
+        let commands = event.content.body;
+
+        try {
+            // Seperate [server+commands, ...]
+
+            let matchBlocks = commands.match(/(\r?\n)?([a-z0-9\.:-]+)(\r?\n([A-Z]+( .*)?))+/g);
+            // Each block looks like this:
+            // irc.server
+            // (COMMAND [arg0 [arg1 [...]]])
+            // (COMMAND [arg0 [arg1 [...]]])
+            // ...
+
+            if (!matchBlocks) {
+                throw new Error(`Badly formed command block(s)`);
+            }
+
+            for (let i = 0; i < matchBlocks.length; i++) {
+                let block = matchBlocks[i];
+
+                let ircDomain = block.match(/^(?:\r?\n)?([a-z0-9\.:-]+)/)[1];
```

Comment:
This looks to be the same as the regex in `matchBlocks` for the first line. Likewise on `:423`. I would try to keep this simple. Regular expressions are hard to read, and what you're doing doesn't require complicated matches. Why not just something like:
```javascript
var lines = commands.replace("\r", "").split("\n"); // remove \r and split on the remaining \n's
var currentDomain = null;
lines.forEach((line) => {
  // regex for matching server
  // else regex for matching command
  // else throw
});
```